slack/rtm/client.py

```diff
@@ -357,18 +366,42 @@ def _next_msg_id(self):
                 raise
 
     async def _read_messages(self):
-        """Process messages received on the WebSocket connection.
-
-        Iteration terminates when the client is stopped or it disconnects.
-        """
+        """Process messages received on the WebSocket connection."""
         while not self._stopped and self._websocket is not None:
-            async for message in self._websocket:
-                if message.type == aiohttp.WSMsgType.TEXT:
-                    payload = message.json()
-                    event = payload.pop("type", "Unknown")
-                    self._dispatch_event(event, data=payload)
-                elif message.type == aiohttp.WSMsgType.ERROR:
-                    break
+            try:
+                # Wait for a message to be recieved, but timeout after a second so that
+                # we can check if the socket has been closed, or if self._stopped is
+                # True
+                message = await self._websocket.receive(timeout=1)
+            except asyncio.TimeoutError:
+                if not self._websocket.closed:
```

Comment:
I'd love to see some unit tests added for this condition.

Reply:
I've spent some time trying to mock up enough to get this tested but I've come up short I'm afraid, and I need to get on with some work. Do you have any bright ideas about how to test this section?